controllers/flowcollector_controller_certificates_test.go

```diff
@@ -486,15 +484,17 @@ func flowCollectorCertificatesSpecs() {
 			}, timeout, interval).Should(Succeed())
 		})
 
-		It("Should not get CR", func() {
-			Eventually(func() error {
-				err := k8sClient.Get(ctx, crKey, &flowCR)
-				if err == nil {
-					err = fmt.Errorf("CR is still present. Status: %s", status.ConditionsToString(flowCR.Status.Conditions))
-				}
-				return err
-			}, timeout, interval).Should(MatchError(`flowcollectors.flows.netobserv.io "cluster" not found`))
-		})
+		// Flaky assertion here, quite annoying, some weirdness with `envtest` maybe?
+		// Anyway it doesn't bring much value. Disabling it for the time being
```

Comment:
I wonder if this is not tied to the parallelization of the tests. Currently, we are not setting any parameter wich means tests from different packages can run concurently.
```
	-parallel n
	    Allow parallel execution of test functions that call t.Parallel, and
	    fuzz targets that call t.Parallel when running the seed corpus.
	    The value of this flag is the maximum number of tests to run
	    simultaneously.
	    While fuzzing, the value of this flag is the maximum number of
	    subprocesses that may call the fuzz function simultaneously, regardless of
	    whether T.Parallel is called.
	    By default, -parallel is set to the value of GOMAXPROCS.
	    Setting -parallel to values higher than GOMAXPROCS may cause degraded
	    performance due to CPU contention, especially when fuzzing.
	    Note that -parallel only applies within a single test binary.
	    The 'go test' command may run tests for different packages
	    in parallel as well, according to the setting of the -p flag
	    (see 'go help build').
```

Adding `-p 1` to the `go test` command ensure each tests to be run separately but highly increase the testing time.
We could also try to use the `t.Parallel` call in each test checking for `Cleanup`
Lastly, we can also force different FlowCollector instance name. It's hacky but may work here.

Reply:
FYI [this PR](https://github.com/netobserv/network-observability-operator/pull/510
) moved FLP and Monitoring controller tests outside of `controllers` package so we parallelize creation / deletion of `FlowCollector` between the 3 packages.

Reply:
So, after many tests, I didn't find issues with parallelism (although I don't guarantee that there aren't any). My guess is that it's ok with parallelism because each controller set up its own envtest environment so they run in isolation. And tbh if this isn't the case, I'm quite surprised to not see many more failures at random places, whereas here this is always the same single test that is failing.
Anyway my understanding of this test failure is finally that this is a timing issue, the CR taking sometimes more time to be actually deleted than the test timeout. So I have this PR for a fix: https://github.com/netobserv/network-observability-operator/pull/557 (and I'll remove my change here)

Reply:
I reverted the change here, cf the mentioned PR above